### PR TITLE
Grid support

### PR DESCRIPTION
**Pull request by golgher**
Added methods to generate Bootstrap grid.

Code example:
```php
echo $this->Html->gridContent($this->Form->input('field1'))
    ->gridConfig(['size' => 4])
    ->gridContent($this->Form->input('field2'))
    ->gridConfig(['size' => 4])
    ->gridContent($this->Form->input('field3'))
    ->gridConfig(['size' => 4])
    ->gridRender();
```

### CODE REVIEW

File: src/View/Helper/HtmlHelper.php [hunk 1]
```diff
@@ -5,6 +5,12 @@ class HtmlHelper extends \Cake\View\Helper\HtmlHelper
 {
     use OptionsAwareTrait;
 
+    private $grid = [];
+
+    private $gridOptions = [];
+
+    private $gridCounter = 0;
```
Comment (dereuromark) [inline src/View/Helper/HtmlHelper.php]: Does it have to be private? Often times people extend this HtmlHelper and would have to redeclare private stuff.
Protected should suffice IMO.

**Comment by dakota** [comment]
I like the idea, but the interface looks very alien and "uncakelike". I'd rather see something like `tableCells`.

**Comment by dereuromark** [comment]
Wouldnt this maybe be an own GridHelper instead of putting all that into the Html one?

File: src/View/Helper/HtmlHelper.php (hunk 1 above)
Comment (dakota) [inline src/View/Helper/HtmlHelper.php]: I would say that the FoC plugins should follow the same standards of core CakePHP where no private methods or properties are allowed.

**Comment by ADmad** [comment]
- Don't use `private` properties or methods
- Don't uses HTML fragments like in `gridRender()`. Either use string templates and/or elements.
- As @dereuromark sadi better to have a separate helper instead of adding more methods to HtmlHelper

File: src/View/Helper/HtmlHelper.php (hunk 1 above)
Comment (jippi) [inline src/View/Helper/HtmlHelper.php]: :+1:

**Comment by golgher** [comment]
I agree with almost everything, I just think it should be part of the Html Helper because it will render html code, but that's my opinion.
I think the project could implement other Bootstrap's components inside the HtmlHelper, such as: Navs (tabs and pills), List Group, Panels, Wells, Grid, etc.

**Comment by dereuromark** [comment]
Almost every helper generates HTML :) See Form or other helpers.
The question is more: Is it a large enough subpart to be its own helper or is it OK to through into the existing one.
If you say "Navs (tabs and pills), List Group, Panels, Wells, Grid, etc" then this sounds to me like a new separate one rather than bloating the HtmlHelper IMO.

**Comment by golgher** [comment]
Sorry, I think I explained wrong. I think the HtmlHelper of the Bootstrap's project, should know how to write the html components of Bootstrap.

**Comment by lorenzo** [comment]
@golgher Having it there makes sense to me

**Comment by jadb** [comment]
I don't think the `HtmlHelper` should be bloated with `grid` nor `components`. I started things in here because it was simpler but if there is a need to add other features, they are complex domains that would require their own respective helpers IMO.

**Comment by ADmad** [comment]
Closing this due to lack of activity. 

@golgher You are welcome to reopen or make a new PR with changes as requested.

**Pull request closed without merging**